src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -553,13 +573,13 @@ ResponseCode RTCPMessageManager::processBindConnectionResponse(
 }
 
 ResponseCode RTCPMessageManager::processCheckLogicalPortsResponse(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.

Reply:
It's not, its a reference of an object. And while this object is alive, the pointer is not freed.